examples/java/src/main/java/org/apache/beam/examples/snippets/transforms/io/gcp/bigquery/BigQueryReadFromQueryWithBigQueryStorageAPI.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.examples.snippets.transforms.io.gcp.bigquery;
+
+// [START bigquery_read_from_query_with_bigquery_storage_api]
+
+import java.util.Arrays;
+import org.apache.beam.examples.snippets.transforms.io.gcp.bigquery.BigQueryMyData.MyData;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.TypedRead.Method;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+
+class BigQueryReadFromQueryWithBigQueryStorageAPI {
+  public static PCollection<MyData> readFromQueryWithBigQueryStorageAPI(
+          String project, String dataset, String table, String query, Pipeline pipeline) {
+
+    // String project = "my-project-id";
+    // String dataset = "my_bigquery_dataset_id";
+    // String table = "my_bigquery_table_id";
+
+    // Pipeline pipeline = Pipeline.create();
+
+    /*
+    String query = String.format("SELECT\n" +
+        "  string_field,\n" +
+        "  int64_field,\n" +
+        "  float64_field,\n" +
+        "  numeric_field,\n" +
+        "  bool_field,\n" +
+        "  bytes_field,\n" +
+        "  date_field,\n" +
+        "  datetime_field,\n" +
+        "  time_field,\n" +
+        "  timestamp_field,\n" +
+        "  geography_field,\n" +
+        "  array_field,\n" +
+        "  struct_field\n" +
+        "FROM\n" +
+        "  `%s:%s.%s`", project, dataset, table)
+    */
+
+    PCollection<MyData> rows =
+        pipeline
+            .apply(
+                "Read from BigQuery table",
+                BigQueryIO.readTableRows()
```

Comment:
I would avoid using readTableRows in an example snippet, both for the storage API and also for the existing export-based model -- this involves a needless conversion from Avro to JSON, where customers should instead be able to consume the Avro GenericRecords directly.

Reply:
Okay, agree. What would be prefered way to continue with this then?

Finish this PR with using TableRows to have all 3 examples using the same undesired `readTableRows() call

or do I refactor this only example to use `from(SerializableFunction<SchemaAndRecord, T> f)` as a part of this PR

or do we immediately refactor all 3 examples
- `Reading from a table`
- `Reading with a query string`
- `Using the BigQuery Storage API`

using the preferred API?

Reply:
If you have the cycles, let's do (3). Otherwise, you can go ahead with (1) and I will take care of updating them when you're done.

Reply:
Then let's merge this, and next week I can refactor all 3 examples.